sandy/gls.py

```diff
@@ -60,9 +60,10 @@ def gls_update(x_prior, S, Vx_prior, Vy_extra, y_extra,
 
     """
     # Model calculus:
+    x_prior_ = pd.Series(x_prior)
     S_ = pd.DataFrame(S)
+    S_ = S_.loc[:, x_prior.index]
     y_calc_ = _y_calc(x_prior, S)
-    x_prior_ = pd.Series(x_prior)
     y_extra_ = pd.Series(y_extra)
     y_calc_ = y_calc_.reindex(y_extra_.index)
     S_ = S_.loc[y_extra_.index, :]
```

Comment:
why `S_ = S_.loc[:, x_prior.index]` and `S_ = S_.loc[y_extra_.index, :]` are on two different lines?
Could it be `S_ = S_.loc[y_extra_.index, x_prior.index]`?

Reply:
They are in different lines because i dont want to lose accuracy: In our specific problem, with S_ = S_.loc[y_extra_.index, x_prior.index] the y_calc are more different from the libraries that if we do in separate lines, because we are using more information of qmatrix to calculate y_calc